### PR TITLE
:tada: Add prompt before exiting the program

### LINKED ISSUE #216: Exiting Pyflow should prompt "Exit without saving?"
Exiting Pyflow currently is too easy. We should prompt the user if he wants to exit without saving to avoid the loss of work

### PR DESCRIPTION
**Pull request by FabienRoger**
Fixes #216

### CODE REVIEW

File: pyflow/graphics/window.py [hunk 1]
```diff
@@ -451,46 +453,65 @@ def onEditRun(self):
...
+        msgbox.setCheckBox(cb)
+        msgbox.exec()
+
+        if msgbox.checkBox().checkState() == Qt.CheckState.Checked:
+            self.never_show_exit_prompt = True
+
+        if msgbox.result() == int(str(QMessageBox.No)):
+            event.ignore()
+            return
+
```
Comment (MathisFederico) [inline pyflow/graphics/window.py]: This should replace self.maybeSave

File: pyflow/graphics/window.py [hunk 2]
```diff
@@ -451,46 +453,65 @@ def onEditRun(self):
...
+    #     """Ask for save and returns if the file should be closed.
 
-        Returns:
-            True if the file should be closed, False otherwise.
+    #     Returns:
+    #         True if the file should be closed, False otherwise.
 
-        """
-        if not self.isModified():
-            return True
```
Comment (MathisFederico) [inline pyflow/graphics/window.py:478]: Message shouldn't pop if file is not modified still !

**Review by MathisFederico — CHANGES_REQUESTED** [review]
A few changes and we are good to go !
Thanks for the bugfix !

File: pyflow/graphics/window.py (hunk 1 above)
Comment (FabienRoger) [reply pyflow/graphics/window.py]: I feel like maybeSave is not exactly what we want. First, it doesn't have the checkbox, (and I really don't want to have the exit prompt when trying out the new features I am coding). Second, it isn't clear what should be saved, and how, when there are multiple open widgets, especially when some of them have never been saved.

I believe maybeSave is an outdated function that only worked when it was developed, when there was only one single widget.

Comment (MathisFederico) [reply pyflow/graphics/window.py]: True

Comment (MathisFederico) [reply pyflow/graphics/window.py]: But we should delete it and not comment it then !

File: pyflow/graphics/window.py [hunk 3]
```diff
@@ -451,46 +453,77 @@ def onEditRun(self):
...
         if self.mdiArea.currentSubWindow():
             event.ignore()
         else:
             self.writeSettings()
             event.accept()
 
-    def maybeSave(self) -> bool:
-        """Ask for save and returns if the file should be closed.
+    # def maybeSave(self) -> bool:
+    #     """Ask for save and returns if the file should be closed.
```
Comment (MathisFederico) [inline pyflow/graphics/window.py]:
```suggestion
```
We should delete it I guess

**Review by MathisFederico — CHANGES_REQUESTED** [review]
Just remove maybeSave and we are good to go!

**Review by MathisFederico — APPROVED** [review]
Nice ! Thanks for the bugfix !

**Pull request merged**